helm/cloud-provider-vsphere/charts/cloud-provider-for-vsphere/templates/secret_config.yaml

```diff
@@ -1,23 +1,26 @@
 {{- if .Values.config.enabled | default .Values.global.config.enabled -}}
 apiVersion: v1
-kind: ConfigMap
+kind: Secret
 metadata:
-  name: vsphere-cloud-config
+  name: {{ .Values.config.name | default "cloud-config" }}
   labels:
     app: {{ template "cpi.name" . }}
     vsphere-cpi-infra: cloud-config
     component: cloud-controller-manager
-    {{- include "labels.common" $ | nindent 4 }}
   namespace: {{ .Release.Namespace }}
-data:
+stringData:
   vsphere.conf: |
     # Global properties in this section will be used for all specified vCenters unless overriden in VirtualCenter section.
     global:
       port: 443
+      {{- if .Values.config.tlsThumbprint }}
+      tlsThumbprint: {{- .Values.config.tlsThumbprint }}
+      {{- else }}
       # set insecure-flag to true if the vCenter uses a self-signed cert
       insecureFlag: true
```

Comment:
We should not assume this value to always be true, most big customers will have valid certificates installed on their vCenters. Sounds better to make it configurable with `default: false`?

Reply:
Yeah. You are right.

I found this in the provider chart https://github.com/kubernetes/cloud-provider-vsphere/blob/master/charts/vsphere-cpi/templates/configmap.yaml#L16

And also I found there is a config in the cluster repo too https://github.com/giantswarm/cluster-vsphere/blob/main/helm/cluster-vsphere/values.yaml#L71

I added it to this repo: https://github.com/giantswarm/cloud-provider-vsphere-app/pull/31/files#diff-2c4c70a148aa47d79b8d3a24bae0247c998f22ecfe9e30e8ae59d4ee9c812ac4R12

Reply:
Sorry I didn't specify, I meant `insecureFlag: true`
But it's also good to parameterise the thumbprint.